core/src/com/unciv/ui/tilegroups/CityButton.kt

```diff
@@ -31,25 +30,334 @@ import com.unciv.ui.utils.extensions.colorFromRGB
+
+        val textColor = city.civInfo.nation.getInnerColor()
+
+        val table = Table()
+
+        var growthPercentage = city.population.foodStored / city.population.getFoodToNextPopulation().toFloat()
+        if (growthPercentage < 0) growthPercentage = 0.0f
+        if (growthPercentage > 1) growthPercentage = 1.0f
+
+        val growthBar = ImageGetter.getProgressBarVertical(4f, 30f,
+            if (city.isStarving()) 1.0f else growthPercentage,
+            if (city.isStarving()) Color.RED else CityButton.ColorGrowth, Color.BLACK, 1f)
+        growthBar.color.a = 0.8f
+
+        val turnLabelText = when {
+            city.isGrowing() -> {
+                val turnsToGrowth = city.population.getNumTurnsToNewPopulation()
+                if (turnsToGrowth != null && turnsToGrowth < 100) turnsToGrowth.toString() else "∞"
+            }
+            city.isStarving() -> {
+                val turnsToStarvation = city.population.getNumTurnsToStarvation()
+                if (turnsToStarvation != null && turnsToStarvation < 100) turnsToStarvation.toString() else "∞"
+            }
+            else -> "∞"
+        }
+
+        val turnLabel = turnLabelText.toLabel(fontColor = textColor, fontSize = 13)
+
+        table.add(growthBar).padRight(2f)
+        table.add(turnLabel).expandY().bottom()
+        add(table).minWidth(6f).padLeft(2f)
+    }
+
+    private fun addCityText(city: City, forPopup: Boolean) {
+
+        val textColor = city.civInfo.nation.getInnerColor()
+        val table = Table().apply { isTransform = false }
+
+        if (city.isCapital()) {
+            val capitalIcon = when {
+                city.civInfo.isCityState() -> ImageGetter.getNationIcon("CityState")
+                    .apply { color = textColor }
+                else -> ImageGetter.getImage("OtherIcons/Capital")
+            }
+            table.add(capitalIcon).size(20f).padRight(5f)
+        }
+
+        val cityName = city.name.tr().toLabel(fontColor = textColor, alignment = Align.center)
+        table.add(cityName).growY().center()
+
+        if (!forPopup) {
+            val cityReligion = city.religion.getMajorityReligion()
+            if (cityReligion != null) {
+                val religionImage = ImageGetter.getReligionIcon(cityReligion.getIconName()).apply {
+                    color = textColor }.toGroup(20f)
+                table.add(religionImage).size(20f).padLeft(5f)
+            }
+        }
+
+        table.pack()
+        add(table)
+            .minHeight(34f)
+            .padLeft(10f)
+            .padRight(10f)
+            .expandY().center()
+
+    }
+
+    private fun addCityConstruction(city: City) {
+
+        val textColor = city.civInfo.nation.getInnerColor()
+
+        val cityConstructions = city.cityConstructions
+        val cityCurrentConstruction = cityConstructions.getCurrentConstruction()
+
+        val progressTable = Table()
+
+        var percentage = 0f
+        var turns = "∞"
```

Comment:
Reminds me that this character should probably be a constant but that's a random thought and not a critique.